### PR TITLE
Refactor query and pool tests to use a shared setup

### LINKED ISSUE #10: Improve testing infra
Waiting for `tearDown` and `setUp` from denoland/deno_std#128

Then we need to prepare methods to quickly setup new tables

### PR DESCRIPTION
**Pull request by hayd**
This means they can each be run independently.

fixes #10

### CODE REVIEW

**Comment by hayd** [comment]
@bartlomieju bump!

**Comment by bartlomieju** [comment]
@hayd I'll get to it on Saturday

File: connection.ts [hunk 1]
```diff
@@ -562,15 +568,20 @@ export class Connection {
   }
 
   async initSQL(): Promise<void> {
+    // validate the connection using a
```
Comment (bartlomieju) [inline connection.ts]: I believe this comment is not finished 😅

File: pool.ts [hunk 2]
```diff
@@ -19,12 +18,7 @@ export class Pool {
   }
 
   private async newConnection(): Promise<Connection> {
```
Comment (bartlomieju) [inline pool.ts]: please rename this method to `_createConnection`

File: query.ts [hunk 3]
```diff
@@ -86,8 +86,4 @@ export class Query {
     const encodingFn = config.encoder ? config.encoder : encode;
     return config.args.map(encodingFn);
   }
-
-  async execute(connection: Connection): Promise<QueryResult> {
```
Comment (bartlomieju) [inline query.ts]: Fine 👍

**Comment by bartlomieju** [comment]
I'll land this anyway and fix later, thanks again @hayd

File: query.ts (hunk 3 above)
Comment (hayd) [reply query.ts]: Apologies for missing these comments

**Pull request merged**